server/service/src/sync/integrate_document.rs

```diff
@@ -10,15 +12,31 @@ use crate::programs::{
     patient::{patient_schema::SchemaPatient, patient_updated::update_patient_row},
     program_enrolment::program_enrolment_updated::update_program_enrolment_row,
     program_enrolment::program_schema::SchemaProgramEnrolment,
+    update_program_document::update_program_events,
 };
 
 pub(crate) fn sync_upsert_document(
     con: &StorageConnection,
     document: &Document,
 ) -> Result<(), RepositoryError> {
-    // TODO comment why only insert here
+    // Fetch current document by name to check if the new document is the latest in the DB
+    let current_doc = DocumentRepository::new(con)
+        .query_by_filter(DocumentFilter::new().name(StringFilter::equal_to(&document.name)))?
+        .pop();
+    let new_doc_is_latest = if let Some(current_doc) = current_doc {
+        current_doc.datetime <= document.datetime
+    } else {
+        true
+    };
```

Comment:
I fetch the current doc before inserting the new document to avoid handling the impossible error case that there is no document found after inserting one...